crates/l1tx/src/filter/types.rs

```diff
@@ -1,20 +1,85 @@
+use std::cmp::Ordering;
+
+use bitcoin::{Amount, OutPoint};
 use borsh::{BorshDeserialize, BorshSerialize};
 use strata_primitives::{
     block_credential::CredRule,
     buf::Buf32,
-    l1::{BitcoinAddress, Outpoint},
+    l1::{BitcoinAddress, BitcoinScriptBuf, OutputRef},
     params::{DepositTxParams, RollupParams},
     sorted_vec::SortedVec,
 };
+use strata_state::{
+    bridge_state::{DepositEntry, DepositState},
+    chain_state::Chainstate,
+};
 
 use crate::utils::{generate_taproot_address, get_operator_wallet_pks};
 
+// TODO: This is FIXED OPERATOR FEE for TN1
+pub const OPERATOR_FEE: Amount = Amount::from_int_btc(2);
+
 #[derive(Clone, Debug, PartialEq, Eq, BorshSerialize, BorshDeserialize)]
 pub struct EnvelopeTags {
     pub checkpoint_tag: String,
     pub da_tag: String,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, BorshSerialize, BorshDeserialize)]
+pub struct ExpectedWithdrawalFulfillment {
+    /// withdrawal address scriptbuf
+    pub destination: BitcoinScriptBuf,
```

Comment:
Why can't this just be a `Vec<u8>` here instead of needing a wrapper for `ScriptBuf` and all of the bloat along with that?  Instances of this type are always well-formed since we generate them deterministically from our chainstate.  We're only using this for comparisons with `ScriptBuf`s, which we look at as plain byte slices anyways.

Reply:
Not important, moved on.